var/lib/istio/envoy/envoy_bootstrap_tmpl.json

```diff
@@ -0,0 +1,419 @@
+{
```

Comment:
why duplicate this? I think we should avoid duplication as much as possible. We learned how hard this was to maintain with istio/install and the old install..

Reply:
The plan is to cleanup a bit - for example SDS-only, will have different config for bootstrap, etc.

The file is difficult to maintain because it has too many ifs and complexity - simpler is better.

Reply:
In my opinion maintaining one complex file is better than maintaining one complex file and one simpler file. This does not just apply to this file either. The injection template is another huge one.

In generally I think its better to refactor the original code to support istiod as much as possible rather than copy-pasting. you have amazingly managed to do all of this without a single modification to exist code, but I think it comes at a high cost

Reply:
If we move ahead with istiod - the complex file will go away. Instead of making it even more complex and then spending time to clean it up - I would rather work on cleaning up without having to worry about breaking the many other cases.

Since values.yaml is gone, and mounted-certs will be gone as well - a lot of the templating will go away.